vortex-roaring/src/boolean/mod.rs

```diff
@@ -1,169 +1,118 @@
-use std::sync::{Arc, RwLock};
-
 use compress::roaring_encode;
-use croaring::{Bitmap, Native};
-use vortex::array::{Array, ArrayKind, ArrayRef};
-use vortex::compress::EncodingCompression;
-use vortex::compute::ArrayCompute;
-use vortex::encoding::{Encoding, EncodingId, EncodingRef};
-use vortex::formatter::{ArrayDisplay, ArrayFormatter};
-use vortex::serde::{ArraySerde, EncodingSerde};
-use vortex::stats::{Stats, StatsSet};
-use vortex::validity::ArrayValidity;
-use vortex::validity::Validity;
-use vortex::{impl_array, ArrayWalker};
+use croaring::{Bitmap, Portable};
+use serde::{Deserialize, Serialize};
+use vortex::array::bool::{Bool, BoolArray};
+use vortex::buffer::Buffer;
+use vortex::stats::ArrayStatisticsCompute;
+use vortex::validity::{ArrayValidity, LogicalValidity};
+use vortex::visitor::{AcceptArrayVisitor, ArrayVisitor};
+use vortex::{impl_encoding, ArrayFlatten, OwnedArray};
 use vortex_error::{vortex_err, VortexResult};
-use vortex_schema::DType;
 use vortex_schema::Nullability::NonNullable;
 
 mod compress;
 mod compute;
-mod serde;
-mod stats;
 
-#[derive(Debug, Clone)]
-pub struct RoaringBoolArray {
-    bitmap: Bitmap,
+impl_encoding!("vortex.roaring_bool", RoaringBool);
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RoaringBoolMetadata {
     length: usize,
-    stats: Arc<RwLock<StatsSet>>,
 }
 
-impl RoaringBoolArray {
-    pub fn new(bitmap: Bitmap, length: usize) -> Self {
-        Self {
-            bitmap,
-            length,
-            stats: Arc::new(RwLock::new(StatsSet::new())),
+impl RoaringBoolArray<'_> {
+    pub fn try_new(bitmap: Bitmap, _length: usize) -> VortexResult<Self> {
+        Ok(Self {
+            typed: TypedArray::try_from_parts(
+                DType::Bool(NonNullable),
+                RoaringBoolMetadata {
+                    // TODO(@jdcasale): wtf -- why is _length wrong? why do we need the plus 1?
+                    length: (bitmap.statistics().cardinality + 1) as usize,
```

Comment:
You can't take the length from the bitmap since it only counts the number of "true" values.